### PR TITLE
Add Trivy github action and remediate [BW-466, BW-476]

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: project/Settings.scala [hunk 1]
```diff
@@ -134,22 +134,21 @@ object Settings {
    */
   val installGcloudSettings: List[Def.Setting[Seq[Instruction]]] = List(
     dockerCustomSettings := List(
-      // instructions to install Google Cloud SDK
-      Instructions.Run("curl https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.tar.gz > /tmp/google-cloud-sdk.tar.gz"),
-      Instructions.Run("""mkdir -p /usr/local/gcloud \
-                         | && tar -C /usr/local/gcloud -xvf /tmp/google-cloud-sdk.tar.gz \
-                         | && /usr/local/gcloud/google-cloud-sdk/install.sh"""
-        .stripMargin),
```
Comment (salonishah11) [inline project/Settings.scala:142]: Out of curiosity, any reason to move these instructions to the end of sequence?

**Review by rsasch — APPROVED** [review]

File: src/ci/docker-compose/cromwell-test/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,8 @@
-FROM openjdk:8
+FROM adoptopenjdk:8-jre-hotspot
```
Comment (aednichols) [inline src/ci/docker-compose/cromwell-test/Dockerfile:1]: I thought we could "adopt" (I don't know the right term) the images created by appsec and build on top of them

Mildly preferable if we don't have to check in and then maintain `apt` commands to Cromwell

File: scripts/docker-develop/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,8 @@
-FROM openjdk:8u252
+FROM adoptopenjdk:8
+
+RUN apt-get update -q && \
+    apt-get upgrade -qq && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (aednichols) [inline scripts/docker-develop/Dockerfile:5]:
```suggestion
FROM us.gcr.io/broad-dsp-gcr-public/base/jre:8-debian
```

File: src/ci/docker-compose/cromwell-test/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,8 @@
...
+
+RUN apt-get update -q && \
+    apt-get upgrade -qq && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (aednichols) [inline src/ci/docker-compose/cromwell-test/Dockerfile:5]:
```suggestion
FROM us.gcr.io/broad-dsp-gcr-public/base/jre:8-debian
```

File: project/Publishing.scala [hunk 5]
```diff
@@ -47,7 +47,10 @@ object Publishing {
       val additionalDockerInstr: Seq[Instruction] = dockerCustomSettings.value
 
       new Dockerfile {
-        from("openjdk:8")
+        from("adoptopenjdk:8-jre-hotspot")
+        runRaw("""apt-get update -q && \
+                 | apt-get upgrade -qq && \
+                 | rm -rf /var/lib/apt/lists/*""".stripMargin)
```
Comment (aednichols) [inline project/Publishing.scala:53]:
```suggestion
        from("us.gcr.io/broad-dsp-gcr-public/base/jre:8-debian")
```

**Review by aednichols — CHANGES_REQUESTED** [review]
I _think_ this is more the spirit of what @dinvlad is trying to achieve: use pre-built images rather than customizing our own

JRE images live [here in GCR](https://console.cloud.google.com/gcr/images/broad-dsp-gcr-public/US/base/jre?project=broad-dsp-gcr-public&folder&organizationId&gcrImageListsize=30)

**Comment by dinvlad** [comment]
@aednichols yup, feel free to use these base images, as they're regularly built and scanned by us - then the GitHub Action will just skip over them.

**Comment by aednichols** [comment]
As a side benefit I think using GCR gets us out of worrying about DockerHub pull limits.

**Comment by cjllanwarne** [comment]
We now re-scan all built images on every PR with the new github actions, so whichever way we go we will have trivy approval on every pull request. That said, I don't have a problem with switching to the pre-scanned images and will make the change (except for the "docker-develop" image which relies on having a JDK (not just a JRE), so the prebuilt images aren't appropriate).

 for the others I'm happy to make the change.

File: src/ci/docker-compose/cromwell-test/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,8 @@
-FROM openjdk:8
+FROM us.gcr.io/broad-dsp-gcr-public/base/jre:8-debian
+
+RUN apt-get update -q && \
+    apt-get upgrade -qq && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (aednichols) [inline src/ci/docker-compose/cromwell-test/Dockerfile:5]:
```suggestion
FROM us.gcr.io/broad-dsp-gcr-public/base/jre:8-debian
```

**Review by aednichols — CHANGES_REQUESTED** [review]
In any location where we're using the `us.gcr.io/broad-dsp-gcr-public/base/jre:8-debian` image I think we don't need the `apt-get` commands because they are already being run when the [image is constructed](https://github.com/broadinstitute/dsp-appsec-blessed-images/blob/main/jre/Dockerfile.8-debian).

If I'm missing something apologies for harping on this twice.

File: src/ci/docker-compose/cromwell-test/Dockerfile (hunk 6 above)
Comment (dinvlad) [reply src/ci/docker-compose/cromwell-test/Dockerfile:5]: Yep you're right!

Comment (cjllanwarne) [reply src/ci/docker-compose/cromwell-test/Dockerfile:5]: They do indeed already get run when the image is constructed but I don't think there's any downside to re-running at our own image build time as well?

 Remember that we're also re-running Trivy against the new images we build so we aren't "losing our green tick" and might possibly bring in important new updates that way (eg if the "official" images are only pushed on an infrequent schedule

Comment (aednichols) [reply src/ci/docker-compose/cromwell-test/Dockerfile:5]: a.k.a. DRY

Comment (dinvlad) [reply src/ci/docker-compose/cromwell-test/Dockerfile:5]: I agree - and we update those images weekly. We can also switch to a daily schedule if you think that'd be helpful. Though if that's not good enough, then please feel free to just use the `adoptopenjdk:8` image - that's as good as our "blessed" ones, but we just wanted to simply that part of work for you 😉

Comment (cjllanwarne) [reply src/ci/docker-compose/cromwell-test/Dockerfile:5]: 🤷‍♂️  I don't feel particularly strongly so I'll switch it over to "just use the base image"

**Review by aednichols — APPROVED** [review]
LGTM 👍 @cjllanwarne and thank you @dinvlad

**Pull request merged**